Lib/profiling/sampling/sample.py

```diff
@@ -675,10 +679,13 @@ def _validate_collapsed_format_args(args, parser):
 
 def wait_for_process_and_sample(pid, sort_value, args):
     """Sample the process immediately since it has already signaled readiness."""
-    # Set default collapsed filename with subprocess PID if not already set
+    # Set default filename with subprocess PID if not already set
     filename = args.outfile
-    if not filename and args.format == "collapsed":
-        filename = f"collapsed.{pid}.txt"
+    if not filename:
+        if args.format == "collapsed":
+            filename = f"collapsed.{pid}.txt"
+        elif args.format == "gecko":
```

Comment:
ℹ️ **Maintainability**

Consider adding a validation function for the `gecko` format similar to `_validate_collapsed_format_args`. This will help maintain consistency and ensure that all necessary arguments for the `gecko` format are validated.